From dtor@insightbb.com Tue Aug  1 19:33:39 2006
From: Dmitry Torokhov <dtor@insightbb.com>
To: linux-usb-devel <linux-usb-devel@lists.sourceforge.net>
Subject: USB: Make usb_buffer_free() NULL-safe
Date: Tue, 1 Aug 2006 22:33:34 -0400
Cc: Greg KH <gregkh@suse.de>
Content-Disposition: inline
Message-Id: <200608012233.34752.dtor@insightbb.com>

kfree() handles NULL arguments which is handy in error handling paths as one
does need to  insert bunch of ifs. How about making usb_buffer_free() do the
same?

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/usb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/usb.c
+++ gregkh-2.6/drivers/usb/core/usb.c
@@ -761,7 +761,9 @@ void usb_buffer_free (
 )
 {
 	if (!dev || !dev->bus || !dev->bus->op || !dev->bus->op->buffer_free)
-	    	return;
+		return;
+	if (!addr)
+		return;
 	dev->bus->op->buffer_free (dev->bus, size, addr, dma);
 }
 
